wodles/aws/aws_s3.py

```diff
@@ -3044,6 +3058,23 @@ def arg_valid_regions(arg_string):
     return final_regions
 
 
+def arg_valid_iam_role_duration(duration: int or None, iam_role_arn: str or None):
+    """Checks if the role session duration specified is a valid parameter.
+
+    Parameters
+    ----------
+    duration: int or None
+        The desired session duration in seconds.
+    iam_role_arn: str or None
+        The iam role arn provided.
+    """
+    # Session duration must be between 15m and 12h
+    if not (duration is None or (900 <= duration <= 3600)):
+        raise ValueError("Invalid session duration specified. Value must be between 900 and 3600.")
+    if duration is not None and iam_role_arn is None:
```

Comment:
Remove this. As you cannot access the `iam_role_arn` value from here you will need to move this condition to the `main` body. See the next comment for more details.